### PR TITLE
Add German Translations

### PR DESCRIPTION
**Pull request by ffried**
... and a `.gitignore`.

The German translation is probably not final as-is. There are many aspects where I either had no idea in what context they are used, or had no idea about what it actually means at all (and couldn't find anything - probably also due to lack of context).
So it would be good if you can find someone to look over it and correct any mistakes. But it should serve as a good starting point with many obvious texts being done. I'd expect the changes to only be within the nitty gritty details.

I'd be happy to look through it, if there's a test version with the German translation, though.

<hr/>

**Note:** In German as used in Germany and Austria, there is a the `ß` character, that is used in some places to mark a different pronunciation of `ss`. This is not used in Switzerland and Liechtenstein (which also use German), however. I initially planned to duplicate the translation for the former countries, adding `ß` where needed, but it's actually easier the other way round.

### CODE REVIEW

**Comment by HarshilShah** [comment]
Hey Florian, thank you so much for this PR! I've asked on Twitter/Mastodon/etc. to see if anyone else can give it a look, too!

Please let me know what the ambiguities are, I'd love to add comments and improve the experience for translators!

As for units, Apple does provide some really good APIs for formatting `Measurement`s, but they are a bit limited in terms of how you can extend them for things like steps, workouts, etc., and you also can't extract the correctly pluralised unit text alone, which is useful for contexts like text fields, where only the number is editable, and the unit needs to be displayed separately, and so I've added manual localisation strings for those cases

**Comment by HarshilShah** [comment]
Just adding an Xcloc export of this branch so folks reviewing it can open it without downloading Xcode: [de.xcloc.zip](https://github.com/HarshilShah/PeakLocalisations/files/13945633/de.xcloc.zip)

File: PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings [hunk 1]
```diff
@@ -32,6 +51,12 @@
       "comment" : "Privacy - Health Update Usage Description",
       "extractionState" : "extracted_with_value",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Peak schreibt dein Gewicht und deine Körpermasse in HealthKit"
```
Comment (dhoepfl) [inline PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:57]: Should be “Körpermaße”.

**Comment by ffried** [comment]
I've just revised a few translations that came to me over night (also fixed a few mistakes I spotted when looking through them again)...

Also, here's a few examples of where translating is difficult (including reasons):

- `Formatting`: A lot of duplicated texts, that are essentially all the same. Some of which I understand, some I think could be simplified (e.g. using the attributed variants in non-attributed places, by not adding any attributes?). Also, it might make sense to add a comment to the attributed variants, explaining how they are structured - might look cryptic to people not familiar with `AttributedString`...
- `AddEntry`: `Log %@` -> That's a tricky one. Your English translation says "Log Your %@". However, the translation of "Your" in German might vary depending on what is inserted for `%@`.
- `Details`: `AddBlock.Title` -> It would be helpful to know what is inserted for `%@` - even though it likely won't affect the German translation here. Same for `RemoveBlock.Title`.
-  `PeakPro`: `Reason.Themes` and `Reason.Widgets` -> These look like prefixes, but what coming after? (I assumed "Peak Pro").
- `Widget`: `%@ Hold to set up` and `%@ Tap and hold to set up`: A comment describing what is inserted for `%@` would help
- `Widget`: `None` -> The German translation could actually vary (e.g. between "Nichts" or "Keine") based on what "None" means here. In this case it seems to be related to "Metrics" (as in "No Metric"). Also note, that in case of "Keine", it is even more related to the context it is used in, since there are variations ("Keines", "Keiner", ...).
- `Goals`: `Stat.DaysInHand`: Not sure in what context this is used in.

I think that's most of what I struggled with.

Another thing that's tricky with German (but likely other languages as well), is that there might be multiple translations for one word - especially words that are nowadays also understood in English. One example is "Theme": you could translate it as "Stil", but "Theme" itself is also understood (and in this case probably more appropriate, which is why I just changed that).
This is not a bad thing per se, but given that the word appears in multiple translations (across multiple files) you need to keep in mind what choice you made, so that it's consistent across all files.
Many words are translated multiple times (and I mostly do understand why!), which makes this even harder.

<hr/>
<br/>
To sum it up, apart from the list above, I'm not sure how much can be done to simplify translations, though. I just wanted to share what I noticed during translation.

File: PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings (hunk 1 above)
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings]: Fixed.

**Comment by HarshilShah** [comment]
Thank you so much for these notes! I'll try to add some comments in there to clarify a lot of this!

About the formatting, when I try to instantiate the attributed string key with a regular string, it seems to decode the attributes as regular text, but let me see if that's something I can work around somehow.

As for the `Log` title, that will be weight, BMI, lean body mass, or body fat percentage, basically prefilling in the title from there. Do you think it needs separate strings? And for `Add Block` it'll be Chart, Goal, Benchmark, Trends, etc., so please let me know if that needs separate strings too

> * `Widget`: `%@ Hold to set up` and `%@ Tap and hold to set up`: A comment describing what is inserted for `%@` would help

The substitutions here are an SF Symbol for "hand.point.up.left"

Screenshot 2024-01-16 at 2 16 34 PM

As for `Goals.Stat.DaysInHand`, yeah that's a tricky one for me to explain in English too, but basically it says how many days you have left in the period (week/month). This includes today if you haven't met the goal yet (and so it's still "in hand"), but if you have, today isn't counted.

**Comment by ffried** [comment]
> As for the Log title, that will be weight, BMI, lean body mass, or body fat percentage, basically prefilling in the title from there. Do you think it needs separate strings? 

Then definitively yes. For example, with "Weight" it would be "Trage dein Gewicht ein", but "Log Your Lean Body Mass" would be "Trage deine magere Körpermasse ein". Notice that the "m" in "magere" is lowercase (but uppercase in the title alone), and also it's "deine" vs. "dein".
For the `Add Block`, I've adjusted the German translation so that it shouldn't matter. Though I can imagine other languages not having that luxury...

`Goals.Stat.DaysInHand` -> Ok, then I got that right. Btw. I think it should be "Days At Hand" in English.

The SF Symbol that is inserted is a very important comment. That explains why the text starts with a capital letter (something @dhoepfl also mentioned on Mastodon).

**Comment by ffried** [comment]
To have it documented here as well: https://applelocalization.com is a great resource for checking how Apple translates something. Thanks @dhoepfl for [pointing this out on Mastodon](https://layer8.space/@teilweise/111764759683187961).

I've used it to align many translations with Apple's now. Especially for the workout types.

**Comment by HarshilShah** [comment]
> Then definitively yes. For example, with "Weight" it would be "Trage dein Gewicht ein", but "Log Your Lean Body Mass" would be "Trage deine magere Körpermasse ein". Notice that the "m" in "magere" is lowercase (but uppercase in the title alone), and also it's "deine" vs. "dein".

Gotcha, I've just updated `main` to account for this! It's now 4 separate specialised strings instead of one

**Comment by ffried** [comment]
> Gotcha, I've just updated main to account for this! It's now 4 separate specialised strings instead of one

Great, thanks. Just added the translations for these as well.
I think this looks good now from my side.

File: PeakLocalisations/Localisation/App Store/Metadata.xcstrings [hunk 2]
```diff
@@ -16,6 +22,12 @@
...
... Home- und Sperrbildschrim\nPeak enthält eine Vielzahl wunderschön gestalteter Widgets, mit denen du deine Fortschritte auf einen Blick sehen kannst. So kannst du deine Statistiken für den heutigen Tag, die Fortschritte deiner Ziele und vieles mehr im Auge behalten. Jedes Widget kann mit einem eigenen Theme für den hellen und dunklen Modus angepasst werden.\n\n→ Eine Auswahl an Themes\nPeak enthält über 60 wunderschön gestaltete Themes, damit du genau das Aussehen findest, das deinem Geschmack entspricht. Du kannst sogar verschiedene Themes für den hellen und den dunklen Modus einstellen.\n\n→ Privatsphäre von Anfang an\nDatenschutz ist einer der Grundwerte von Peak. Gesundheitsdaten gehören zu den sensibelsten Daten, die Menschen haben, deshalb gibt es kein Tracking und keine Datenerfassung.\n\n→ Angenehme Bedienung\nPeak fühlt sich auf deinem iPhone und iPad wie zu Hause und enthält vertraute Muster und viele kleine Details, die die Nutzung der App zu einem Kinderspiel machen. Du kannst verschiedene Aspekte anpassen, z. B. das App-Symbol, die Themes, die gewählten Einheiten und vieles mehr.\n\nNutzungsbedingungen: https://www.apple.com/legal/internet-services/itunes/dev/stdeula/"
```
Comment (Manuel-Kehl) [inline PeakLocalisations/Localisation/App Store/Metadata.xcstrings:28]: Since @HarshilShah was asking around for German speakers to do a quick double-check, I had a brief skim and only found one potential typo:

@ffried, I assume you meant to say "Erstelle eine individuelle Übersicht, ~das~ **die** auf deine Bedürfnisse zugeschnitten ist, ...", right?

Apart from this the translation looks great 👍

Comment (ffried) [reply PeakLocalisations/Localisation/App Store/Metadata.xcstrings]: Good catch, thanks! That's a leftover from changing "Dashboard" to "Übersicht"...
Fixed now.

**Comment by HarshilShah** [comment]
I guess if everyone's happy with the translations right now, we're good to merge?

**Comment by HarshilShah** [comment]
And merged! Thank you to everyone for helping with this!

**Comment by ffried** [comment]
@arik-so thanks a lot for your comments! I'm AFK today, but I'll integrate them in the coming days!

File: PeakLocalisations/Localisation/App Store/Metadata.xcstrings [hunk 3]
```diff
@@ -16,6 +22,12 @@
...
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/Metadata.xcstrings:28]: I think the plural is correct here, since it is about multiple workouts, not just one. And even though German imported that word, the plural is still "Trainings".

As for the added `an`, it would change the meaning of the sentence. It's about Peak feeling "right at home" (original) on your iPhone, since it uses the same patterns. It's more in the sense of 'it fits right in'.

File: PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings [hunk 4]
```diff
@@ -53,16 +84,45 @@
...
+        }
+      }
     },
     "Something Wrong?" : {
-
+      "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Etwas stimmt nicht?"
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:101]: Good point, changed.

File: PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings [hunk 5]
```diff
@@ -32,6 +51,12 @@
       "comment" : "Privacy - Health Update Usage Description",
       "extractionState" : "extracted_with_value",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Peak schreibt dein Gewicht und deine Körpermaße in HealthKit"
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:57]: Yea, I mass-changed "ss" to "ß" (see older review comments), and went over it too quickly apparently. Thanks for catching this!

File: PeakLocalisations/Localisation/App Store/Screenshots.xcstrings [hunk 6]
```diff
@@ -59,6 +89,12 @@
     "Resolutions.Review" : {
       "extractionState" : "manual",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Blicke auf deinen Fortschritt 2023 zurück"
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/Screenshots.xcstrings:95]: I'm not a big fan of using only nouns. Using the verb "zurückblicken" sounds more natural IMHO. I added an `in`, because you're not looking at the progress of the year itself, but at the progress _you_ made _within_ that year.

File: PeakLocalisations/Localisation/App Store/Screenshots.xcstrings [hunk 7]
```diff
@@ -103,6 +157,12 @@
     "Widgets.Lockscreen" : {
       "extractionState" : "manual",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "...Und für deinen Sperrbildschirm"
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/App Store/Screenshots.xcstrings:163]: It is a continuation of the previous sentence that still somewhat stands on its own (it's used in screenshots and both texts are used in a separate screenshot where the second follows the first).

image

But your points are valid. Btw. this is the only text that is not using the unicode character in English either, so that's why I kept the three dots.
@HarshilShah is there a specific reason for this here in English or just an oversight?

File: PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings [hunk 8]
```diff
@@ -62,6 +92,12 @@
       "comment" : "Separate instructions with a single new line",
       "extractionState" : "manual",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Drücke lange auf einen freien Bereich auf deinem Sperrbildschirm.\nTippe auf \"Anpassen\".\nWähle den Sperrbildschirm aus.\nWähle ein Rechteck über oder unter der Zeit aus.\nSuche \"Peak\".\nWähle irgendein Widget aus um es hinzuzufügen (Hinweis: Du musst eventuell existierende Widgets entfernen, wenn nicht genug Platz vorhanden ist)\nTippe auf das Widget, um es zu bearbeiten."
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings:98]: I changed it to "Wähle eines der Widgets aus" - sounds a bit more on point to me (since you're presented with a list of widgets at that point).

File: PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings [hunk 9]
```diff
@@ -178,6 +268,12 @@
       "comment" : "Separate instructions with a single new line",
       "extractionState" : "manual",
       "localizations" : {
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Hinweis: Der StandBy Modus funktioniert nur auf einem iPhone mit iOS 17 oder neuer.\nGehe in die Einstellungen > StandBy und stelle sicher, dass StandBy aktiviert ist.\nVerbinde dein iPhone mit einem Ladegerät und kippe es in eine horizontale Position.\nDrücke den Seitenschalter.\nStreiche links oder rechts, um zwischen den verschiedenen Ansichten zu wechseln. Streiche hoch oder runter um zwischen den verschiedenen Optionen der Ansichten zu wechseln."
```
[the remark this answers is not in the document]
Comment (ffried) [reply PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings:274]: I actually changed it to just "StandBy" (as Apple does). Search for "StandBy" on the [official Apple iOS 17 page](https://www.apple.com/de/ios/ios-17/) and click on the plus in the lower right of the picture to see a text.

Similar for the reference to the minimum requirements btw. IMHO it's more natural to reference a single device than multiple ones, since most people only have one (active) iPhone.

File: PeakLocalisations/Localisation/App Store/Screenshots.xcstrings (hunk 7 above)
[the remark this answers is not in the document]
Comment (HarshilShah) [reply PeakLocalisations/Localisation/App Store/Screenshots.xcstrings:163]: Yeah definitely just an oversight, will fix
